File: saerom_j/src/main/java/com/example/saerom_j/login/dto/UserDto.java
package com.example.saerom_j.login.dto;

import com.example.saerom_j.login.entity.User;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

@AllArgsConstructor
@Getter
@Builder

public class UserDto {
    private Long id;
    private String uId;
    private String name;
    private String pwd;
    private String email;

    public static UserDto fromEntity(User user) {
        UserDto userDto = UserDto.builder()
                .id(user.getId())
                .uId(user.getUId())
                .name(user.getName())
                .pwd(user.getPwd())
                .email(user.getEmail())
                .build();
        return userDto;
    }
}

File: SAROM_Back/src/main/kotlin/com/example/sarom_back/account/dto/UserDto.kt
package com.example.sarom_back.account.dto

import com.example.sarom_back.account.entity.UserEntity

data class UserDto(
    var id: Long? = null,
    var name: String? = null,
    var uid: String? = null,
    var pwd: String? = null,
    var email: String? = null
) {
    companion object {
        fun toUserDto(userEntity: UserEntity): UserDto {
            return UserDto(
                id = userEntity.id,
                name = userEntity.name,
                uid = userEntity.uid,
                pwd = userEntity.pwd,
                email = userEntity.email
            )
        }
    }
}